clients/drcachesim/tools/schedule_stats.cpp

```diff
@@ -144,6 +144,29 @@ schedule_stats_t::parallel_shard_exit(void *shard_data)
     return true;
 }
 
+void
+schedule_stats_t::get_scheduler_stats(memtrace_stream_t *stream, counters_t &counters)
+{
+    counters.switches_input_to_input =
+        static_cast<int64_t>(stream->get_schedule_statistic(
+            memtrace_stream_t::SCHED_STAT_SWITCH_INPUT_TO_INPUT));
+    counters.switches_input_to_idle = static_cast<int64_t>(stream->get_schedule_statistic(
+        memtrace_stream_t::SCHED_STAT_SWITCH_INPUT_TO_IDLE));
+    counters.switches_idle_to_input = static_cast<int64_t>(stream->get_schedule_statistic(
+        memtrace_stream_t::SCHED_STAT_SWITCH_IDLE_TO_INPUT));
+    counters.switches_nop = static_cast<int64_t>(
+        stream->get_schedule_statistic(memtrace_stream_t::SCHED_STAT_SWITCH_NOP));
+    counters.quantum_preempts = static_cast<int64_t>(
+        stream->get_schedule_statistic(memtrace_stream_t::SCHED_STAT_QUANTUM_PREEMPTS));
+    counters.migrations = static_cast<int64_t>(
+        stream->get_schedule_statistic(memtrace_stream_t::SCHED_STAT_MIGRATIONS));
+
+    // XXX: If we want to match what "perf" targeting this app would record, we
+    // should remove idle-to-input and add input-to-idle (though generally those two
+    // counts are pretty similar).  OTOH, if we want to match what "perf" systemwide
+    // would record, we would want to add input-to-idle on top of what we have.
```

Comment:
I don't understand this comment.  Don't the stats already include input-to-idle?  Why would we need to add it?

Reply:
The schedule_stats tool is not counting input-to-idle today in its total_switches count (or any of its counts).  idle-to-input is very close to input-to-idle so its result is close enough to a targeted perf.

Reply:
I.e, the schedule_stats code is counting swap-ins for a non-idle incoming thread, while the kernel event counts swap-outs of a targeted thread.

Reply:
Elaborated in the comment here, and put a comment in the counting code.